src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/GetAllocationResourceIntTest.kt

```diff
@@ -817,6 +974,266 @@ class GetAllocationResourceIntTest : IntegrationTestBase() {
       .exchange()
       .expectStatus().isOk
 
+  /*
+   * This endpoint is closely related to `GET /allocations/ids` and any changes to the allocation selection rules should
+   * affect it. For this reason I've added an allocation with a missing pay band to each of the tests in the `AllocationSelection`
+   * nested class above to further test this endpoint.
+   */
+  @Nested
+  @DisplayName("GET /allocations/missing-pay-bands")
```

Comment:
Tests the various edge cases where a specific allocation does or doesn't have a missing pay band.